File: implementations/php/pcf-dcp/src/Chunker.php
<?php

declare(strict_types=1);

namespace Kduma\PCFDCP;

/**
 * How a Writer splits an inner partition's content into extents (spec Section
 * 10.2; chunking is writer-side policy).
 */
final class Chunker
{
    private function __construct(
        public readonly bool $isWhole,
        public readonly int $size,
    ) {
    }

    /** One extent for the whole content. */
    public static function whole(): self
    {
        return new self(true, 0);
    }

    /** Fixed-size chunks of $n bytes (0 = whole). */
    public static function fixed(int $n): self
    {
        return new self($n <= 0, $n);
    }
}

File: implementations/dotnet/pcf-dcp/src/Pcf.Dcp/Chunker.cs
namespace Pcf.Dcp;

/// <summary>
/// How a Writer splits an inner partition's content into extents (spec Section
/// 10.2; chunking is writer-side policy).
/// </summary>
public sealed class Chunker
{
    /// <summary>Whether this chunker emits one extent for the whole content.</summary>
    public bool IsWhole { get; }

    /// <summary>Fixed chunk size in bytes (meaningful only when not whole).</summary>
    public int Size { get; }

    private Chunker(bool whole, int size)
    {
        IsWhole = whole;
        Size = size;
    }

    /// <summary>One extent for the whole content.</summary>
    public static Chunker Whole() => new Chunker(true, 0);

    /// <summary>Fixed-size chunks of <paramref name="n"/> bytes (0 = whole).</summary>
    public static Chunker Fixed(int n) => new Chunker(n <= 0, n);
}
